internal/plugin6/grpc_provider.go

```diff
@@ -1478,6 +1479,140 @@ func (p *GRPCProvider) ConfigureStateStore(r providers.ConfigureStateStoreReques
 	return resp
 }
 
+func (p *GRPCProvider) ReadStateBytes(r providers.ReadStateBytesRequest) (resp providers.ReadStateBytesResponse) {
+	logger.Trace("GRPCProvider.v6: ReadStateBytes")
+
+	schema := p.GetProviderSchema()
+	if schema.Diagnostics.HasErrors() {
+		resp.Diagnostics = schema.Diagnostics
+		return resp
+	}
+
+	if _, ok := schema.StateStores[r.TypeName]; !ok {
+		resp.Diagnostics = resp.Diagnostics.Append(fmt.Errorf("unknown state store type %q", r.TypeName))
+		return resp
+	}
+
+	protoReq := &proto6.ReadStateBytes_Request{
+		TypeName: r.TypeName,
+		StateId:  r.StateId,
+	}
+
+	// Start the streaming RPC with a context. The context will be cancelled
+	// when this function returns, which will stop the stream if it is still
+	// running.
+	ctx, cancel := context.WithCancel(p.ctx)
+	defer cancel()
+
+	client, err := p.client.ReadStateBytes(ctx, protoReq)
+	if err != nil {
+		resp.Diagnostics = resp.Diagnostics.Append(grpcErr(err))
+		return resp
+	}
+
+	var buf *bytes.Buffer
+	var expectedTotalLength int
+	for {
+		chunk, err := client.Recv()
+		if err == io.EOF {
+			// End of stream, we're done
+			break
+		}
+		if err != nil {
+			resp.Diagnostics = resp.Diagnostics.Append(err)
```

Comment:
Should this error be wrapped as a gRPC error if it's not an EOF error?

```suggestion
			resp.Diagnostics = resp.Diagnostics.Append(grpcErr(err))
```